src/base/LruMap.h

```diff
@@ -148,39 +169,47 @@ LruMap<EntryValue, EntryCost>::add(const char *key, EntryValue *t)
     if (memLimit() == 0)
         return false;
 
-    index.push_front(new Entry(key, t));
+    auto e = new Entry(key, t, ttl);
+    memUsed_ += e->memCost() + e->key.length() + sizeof(Entry) + sizeof(MapPair);
```

Comment:
Good catch. I have slightly reorganized these. e->memCost() should now account for all the Entry memory usage. The MapPair is a bit more tricky since it is a std::pair object, but using the m->first.length() instead of e->key.length() should more clearly indicate what memory is being accounted for.